.travis.yml

```diff
@@ -1,5 +1,43 @@
+sudo: required
+dist: trusty
+
 language: c
 compiler: gcc
 
+before_script:
+- sudo apt-get -qq install valgrind lcov uuid-dev pkg-config zlib1g-dev libffi-dev gettext libpcre3-dev texinfo gtk-doc-tools cppcheck
+- mkdir cor-dependencies
+- pushd cor-dependencies
+- curl -L -O https://github.com/GNOME/glib/archive/2.49.4.tar.gz
+- tar -xvf 2.49.4.tar.gz
+- pushd glib-2.49.4
+- bash autogen.sh --prefix=/usr
```

Comment:
This is great, but I'd prefer that we put this scripting into a separate file (say `./setup-travis.sh` or something). 

That way, we can add appropriate comments to explain what we're doing and why and we can also use variables to avoid all the hard-coded versions numbers. Travis does support environment variables, but I think a shell script is still going to be more familiar and flexible than taking the Travis approach.

For example, we could do something like this (although a shell function would be even better):

```
glib_version=2.49.4
curl -L -O "https://github.com/GNOME/glib/archive/${glib_version}.tar.gz"
tar -xvf "${glib_version}.tar.gz"
pushd "glib-${glib_version}"
```